### ISSUE TITLE
Orderless matching for `org-set-tags-command`

### ISSUE THREAD

**Issue by clemera** [oantolin/orderless#20]
Hi Omar, you weren't very active on github any more so I hope you are doing well,

I discovered that `orderless` matching does not work with `org-set-tags-command` with multiple tags. You get completion for the first tag only. I have tested it with `emacs -Q` (emacs-version 26.3, org-version 9.3.6)  and `completion-styles` set to `orderless`: 

```ascii
#+tags: some tags to test

* Test                                                                :test:
```
When you press `C-c C-q` with the point on the headline and afterwards press `TAB` inside the minibuffer and you will get `[no match]`. With the default `completion-styles` the completion buffer pops up showing the available tags.

**Comment by oantolin** [comment]
> Hi Omar, you weren't very active on github any more so I hope you are doing well

Hi Clemens! I'm fine, just busy with work. I hope you are doing well too.

This is an interesting sounding bug. I'll see if I can figure it out. I use org mode a lot, but don't really use tags much, so I wouldn't have noticed this myself. Thanks for reporting.

**Comment by clemera** [comment]
I'm fine, too. Just spend to much time in Emacs while I actually wanted to start learning more about Browser APIs :laughing: 

I have looked a bit into it and the problem seems to be that `orderless-filter` let binds `completion-regexp-list` which affects the completion table `org-tags-completion-function` which uses `all-completions` internally (which makes use of `completion-regexp-list`). 

Using something like the following definition of `orderless-filter` fixes it:

```elisp
(defun orderless-filter (string table &optional pred)
  "Split STRING into components and find entries TABLE matching all.
The predicate PRED is used to constrain the entries in TABLE."
  (condition-case nil
      (save-match-data
        (pcase-let* ((`(,prefix . ,pattern)
                      (orderless--prefix+pattern string table pred))
                     (oldcrl completion-regexp-list)
                     (completion-regexp-list
                      (funcall orderless-pattern-compiler pattern))
                     (completion-ignore-case
                      (if orderless-smart-case
                          (cl-loop for regexp in completion-regexp-list
                                   always (isearch-no-upper-case-p regexp t))
                        completion-ignore-case)))
          (all-completions prefix
                           (if (functionp table)
                               (lambda (&rest args)
                                 (let ((completion-regexp-list oldcrl))
                                   (apply table args)))
                             table)
                             pred)))
    (invalid-regexp nil)))
```

**Comment by oantolin** [comment]
I wonder why the `org-set-tags-command` doesn't use `completing-read-multiple` for this. If you change the `completing-read` call to `completing-read-multiple`:

```
(org-trim (completing-read-multiple
				"Tags: "
				org-last-tags-completion-table
				nil nil (org-make-tag-string current-tags)
				'org-tags-history))
```

things seem to work OK.

**Comment by clemera** [comment]
Yeah, the problem is known, see [here](https://gist.github.com/alphapapa/a3433c54a631b693ba9d) which contains some discussion about this, I also wanted to fix it upstream as mentioned there but wasn't motivated enough to proceed.

**Comment by minad** [comment]
I believe this can be closed since it is simply not the fault of Orderless. Substring also does not work. See https://github.com/minad/vertico#org-set-tags-command for some workaround. And one can always use `completing-read-multiple` as a replacement.
